### PR TITLE
[IMP]l10n_ar_account_tax_settlement: retenciones y percepciones IVA txt

### PR DESCRIPTION
**Pull request by pablohmontenegro**
Ticket: 54274

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://runbot.adhoc.com.ar/ingadhoc/odoo-argentina-ee/pull/153).

File: l10n_ar_account_tax_settlement/models/account_journal.py [hunk 1]
```diff
@@ -1343,3 +1344,61 @@ def misiones_files_values(self, move_lines):
...
+                codigo_regimen = line.tax_line_id.codigo_regimen
+                if not codigo_regimen:
+                    raise ValidationError(_('No hay código de régimen en la configuración del impuesto "%s"') % (
+                        line.tax_withholding_id.name))
+                if len(codigo_regimen) < 3:
+                    raise ValidationError(_('El código de régimen tiene que tener 3 dígitos en la configuración del impuesto "%s"') % (line.tax_withholding_id.name))
+                content += codigo_regimen[:3]
+
+                # cuit agente (long 11)
+                line.move_id.partner_id.ensure_vat()
```
Comment (ica-adhoc) [inline l10n_ar_account_tax_settlement/models/account_journal.py]: ensure_vat() return l10n_ar_vat if this exists, call directly on add to content

File: l10n_ar_account_tax_settlement/models/account_journal.py [hunk 2]
```diff
@@ -1343,3 +1344,61 @@ def misiones_files_values(self, move_lines):
...
+                codigo_regimen = payment.tax_withholding_id.codigo_regimen
+                if not codigo_regimen:
+                    raise ValidationError(_('No hay código de régimen en la configuración del impuesto "%s"') % (
+                        payment.tax_withholding_id.name))
+                if len(codigo_regimen) < 3:
+                    raise ValidationError(_('El código de régimen tiene que tener 3 dígitos en la configuración del impuesto "%s"') % (payment.tax_withholding_id.name))
+                content += codigo_regimen[:3]
+
+                # cuit agente (long 11)
+                payment.partner_id.ensure_vat()
```
Comment (ica-adhoc) [inline l10n_ar_account_tax_settlement/models/account_journal.py]: ensure_vat() return l10n_ar_vat if this exists, call directly on add to content

File: l10n_ar_account_tax_settlement/models/account_journal.py [hunk 3]
```diff
@@ -1343,3 +1344,61 @@ def misiones_files_values(self, move_lines):
...
+                # regimen (long 3)
+                codigo_regimen = payment.tax_withholding_id.codigo_regimen
+                if not codigo_regimen:
+                    raise ValidationError(_('No hay código de régimen en la configuración del impuesto "%s"') % (
+                        payment.tax_withholding_id.name))
+                if len(codigo_regimen) < 3:
+                    raise ValidationError(_('El código de régimen tiene que tener 3 dígitos en la configuración del impuesto "%s"') % (payment.tax_withholding_id.name))
+                content += codigo_regimen[:3]
+
+                # cuit agente (long 11)
```
Comment (ica-adhoc) [inline l10n_ar_account_tax_settlement/models/account_journal.py:1364]: the afip file specifies that the length is 13, was it a typing error?

File: l10n_ar_account_tax_settlement/models/account_journal.py [hunk 4]
```diff
@@ -1343,3 +1344,61 @@ def misiones_files_values(self, move_lines):
...
+                # regimen (long 3)
+                codigo_regimen = line.tax_line_id.codigo_regimen
+                if not codigo_regimen:
+                    raise ValidationError(_('No hay código de régimen en la configuración del impuesto "%s"') % (
+                        line.tax_withholding_id.name))
+                if len(codigo_regimen) < 3:
+                    raise ValidationError(_('El código de régimen tiene que tener 3 dígitos en la configuración del impuesto "%s"') % (line.tax_withholding_id.name))
+                content += codigo_regimen[:3]
+
+                # cuit agente (long 11)
```
Comment (ica-adhoc) [inline l10n_ar_account_tax_settlement/models/account_journal.py:1386]: the afip file specifies that the length is 13, was it a typing error?

**Review by ica-adhoc — CHANGES_REQUESTED** [review]
@pablohmontenegro check the comments and prove that functions with the changes requested

File: l10n_ar_account_tax_settlement/models/account_journal.py (hunk 4 above)
Comment (pablohmontenegro) [reply l10n_ar_account_tax_settlement/models/account_journal.py:1386]: I resolve this bacause ica told me that is ok

**Review by ica-adhoc — APPROVED** [review]

**Comment by ica-adhoc** [comment]
@roboadhoc r+

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Comment by fw-bot-adhoc** [comment]
@pablohmontenegro @ica-adhoc this pull request has forward-port PRs awaiting action (not merged or closed):
ingadhoc/odoo-argentina-ee#163

**Pull request closed without merging**